cinema/serializers.py

```diff
@@ -68,13 +81,74 @@ class Meta:
             "movie_title",
             "cinema_hall_name",
             "cinema_hall_capacity",
+            "tickets_available"
         )
 
 
+class TicketRowAndSeatSerializer(serializers.ModelSerializer):
+    row = serializers.IntegerField()
+    seat = serializers.IntegerField()
+
+
 class MovieSessionDetailSerializer(MovieSessionSerializer):
     movie = MovieListSerializer(many=False, read_only=True)
     cinema_hall = CinemaHallSerializer(many=False, read_only=True)
+    taken_places = serializers.SerializerMethodField()
```

Comment:
so here you need to pass your TicketRowAndSeatSerializer and then you will not need your get_taken_seats method